test/connection.test.js

```diff
@@ -1,17 +1,42 @@
 require('./init.js');
 var assert = require('assert');
+var DataSource = require('loopback-datasource-juggler').DataSource;
+var mysqlConnector = require('../');
+var URL = require('url');
 
-var db, DummyModel, odb;
+var db, DummyModel, odb, config;
 
 describe('connections', function () {
 
   before(function () {
     require('./init.js');
 
+    config = global.getConfig();
+
     odb = getDataSource({collation: 'utf8_general_ci', createDatabase: true});
     db = odb;
   });
 
+  it('should pass with valid settings', function (done) {
+    var db = new DataSource(mysqlConnector, config);
+    db.ping(done);
+  });
+
+  it('shoud pass with valid URL override invalid settings', function (done) {
```

Comment:
I think this may be better phrasing: `(it) should pass when valid URL overrides invalid settings`.

On the second thought, the test name should be ideally focused on high-level user perspective, in which case this may be a better name: `(it) ignores all other settings when url is present`.